app/src/main/java/com/asfoundation/wallet/promotions/voucher/VoucherSkuViewHolder.kt

```diff
@@ -0,0 +1,17 @@
+package com.asfoundation.wallet.promotions.voucher
+
+import android.widget.Button
+import androidx.recyclerview.widget.RecyclerView
+import io.reactivex.subjects.PublishSubject
+
+class VoucherSkuViewHolder(private val button: Button) : RecyclerView.ViewHolder(button) {
+
+  fun bind(position: Int, selectedPosition: Int, voucherSkuItem: VoucherSkuItem,
+           onSkuClick: PublishSubject<Int>) {
+    button.text = voucherSkuItem.title
+
+    button.isActivated = selectedPosition == position
+
+    button.setOnClickListener { onSkuClick.onNext(position) }
```

Comment:
The position shouldn't be passed as an argument. You should use `adapterPosition` instead. This is because there are situations with RecyclerViews where this might return the wrong value. For example, let's say you have 5 items in the RV, and you add one more to the 3rd position, so that you have 6. The previous 3rd, 4th and 5th item (which are now 4th, 5th and 6th) will have an outdated position because in this case those 3 will not be rebinded so the click listener is still sending the previous position.